### ISSUE TITLE
Why did the H5 video element full screen control button not respond?

### ISSUE THREAD

**Issue by lyzcw** [fluttercommunity/flutter_webview_plugin#423]
As the title，As shown in the screenshot.
![NoFullsreen](https://user-images.githubusercontent.com/22804961/58077550-5aa1fe00-7bdf-11e9-8c18-eaf55b3da003.jpg)

**Comment by lyzcw** [comment]
As we all know, it's a bad experience to play video on the mobile  without fullscreen.

**Comment by lyzcw** [comment]
@slightfoot @charafau Please pay attention to this Issue.

**Comment by slightfoot** [comment]
Thanks for bringing this up. I think full-screen video playback already works on iOS. This needs confirmation. I am just dropping some reference docs here for later Android implantation. 

https://developer.android.com/reference/android/webkit/WebChromeClient#onShowCustomView(android.view.View,%2520android.webkit.WebChromeClient.CustomViewCallback)
https://developer.android.com/training/system-ui/immersive
https://github.com/cprcrack/VideoEnabledWebView < code not up to stratch but helps understand implentation.

Basically we override onShowCustomView and onHideCustomView to display a Custom-View in-place of the Web-View. We can also choose to enable the immersive mode on Android to hide the system chrome like status. We also want to override the back button behaviour such that if the custom-view is open it will close it on first tap of back button.

Further research required.

**Comment by lyzcw** [comment]
@slightfoot Great. Thank you very much for your response in the middle of the night. Looking forward to the new version.

**Comment by lyzcw** [comment]
> Thanks for bringing this up. I think full-screen video playback already works on iOS. This needs confirmation. I am just dropping some reference docs here for later Android implantation.
> 
> https://developer.android.com/reference/android/webkit/WebChromeClient#onShowCustomView(android.view.View,%2520android.webkit.WebChromeClient.CustomViewCallback)
> https://developer.android.com/training/system-ui/immersive
> https://github.com/cprcrack/VideoEnabledWebView < code not up to stratch but helps understand implentation.
> 
> Basically we override onShowCustomView and onHideCustomView to display a Custom-View in-place of the Web-View. We can also choose to enable the immersive mode on Android to hide the system chrome like status. We also want to override the back button behaviour such that if the custom-view is open it will close it on first tap of back button.
> 
> Further research required.

@slightfoot On Android platform, has this problem been solved?

**Comment by lyzcw** [comment]
@charafau

**Comment by teenstitans** [comment]
Looking forward to solving the problem！

**Comment by mschneider** [comment]
@slightfoot do you see any issues with this solution?

**Comment by lyzcw** [comment]
@slightfoot @charafau Looking forward to the solution of this problem.

**Comment by lyzcw** [comment]
Looking forward to ....

**Comment by lyzcw** [comment]
@slightfoot @charafau Looking forward to the solution of this problem.

**Comment by lyzcw** [comment]
@slightfoot @charafau Looking forward to the solution of this problem.

**Comment by jerry90** [comment]
Looking forward to ....

**Comment by lyzcw** [comment]
@jerry90 https://github.com/fluttercommunity/flutter_webview_plugin/blob/revert-503-mschneider/fullscreen_video/android/src/main/java/com/flutter_webview_plugin/WebviewManager.java，change master branch's Android class to this, sussess!

**Comment by YaseenAbdullah** [comment]
Looking forward to..
